File: app/src/main/java/com/simple_city/app/dto/Purchase.java
package com.simple_city.app.dto;

import com.simple_city.app.entities.*;
import lombok.Data;

import java.util.Set;


@Data
public class Purchase {

    private Customer customer;
    private Address address;
    private Cart cart;
    private Set<CartItemItem> cartItemItems;
}

File: simpleCity-App/simpleCity-app-frontend/src/app/common/purchase.ts
import { Address } from "./address";
import { Cart } from "./cart";
import { CartItemItem } from "./cart-item-item";
import { Customer } from "./customer";

export class Purchase {
    customer!: Customer;
    address!: Address;
    cart!: Cart;
    cartItemItems!: CartItemItem[];
}
